extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -73,8 +75,8 @@ public void handle(AsyncResult<AccessToken> event) {
                     for (String role : OidcUtils.findRoles(clientId, config.getRoles(), token.accessToken())) {
                         builder.addRole(role);
                     }
-                } catch (Exception e) {
-                    result.completeExceptionally(e);
+                } catch (OIDCException e) {
+                    result.completeExceptionally(new ForbiddenException());
```

Comment:
What bugs me a bit is that we don't handle the other exceptions if they happen?

This probably needs to be discussed so will be a bit late for 1.0.1.

Reply:
@gsmet Hi Guillaume, yes, sure. FYI, inside OidcUtils.findRoles there is only unchecked exception which can happen which is `OIDCException` when some paths can not be resolved (all that code does is it iterates over already prepared JSONObject). If something else escapes then I'd consider enhancing `OidcUtils` further. 
However I can also add a RuntimeException catch to catch everything else ? Let me know what you think please, but yes, it can wait

Reply:
I would add the `OIDCException` as the cause of the `ForbuddenException` to not loose the exception context

Reply:
I'm OK with all other exceptions being not handled, they will all result to a ServerError. 
We only want authentication exceptions being transformed to ForbiddenException.

Reply:
Can we keep the previous branch too?

I mean having something like:
```
                } catch (OIDCException e) {
                    result.completeExceptionally(new ForbiddenException());
                } catch (Exception e) {
                    result.completeExceptionally(e);
                }
```

That would have my preference.

Reply:
@gsmet sure

Reply:
@gsmet done